docker/Dockerfile-workers

```diff
@@ -1,39 +1,62 @@
 # syntax=docker/dockerfile:1
-# Inherit from the official Synapse docker image
-ARG SYNAPSE_VERSION=latest
-FROM matrixdotorg/synapse:$SYNAPSE_VERSION
-
-# Install deps
-RUN \
```

Comment:
Is the point that `RUN ...` is reran every time you build the image? (Because there's no way for docker to know if `RUN blah ...` is deterministic and therefore if the resulting layer is cachable?

Reply:
no, the point is that previously we were running the `RUN` on an image based on `matrixdotorg/synapse`, which (of course) changes every time you change a line of code. Since the layer *below* the `RUN` wasn't cached, nor was the layer *including* the `RUN`.

By doing this here on top of a stable base image, the result of the `RUN` is now cached.